File: 02_recursion_basics/02_02_print_1_to_n/Print1ToN.java
public class Print1ToN {
    public static void main(String[] args) {
        int num = 5;
        print_1_to_n(1, num);
    }

    private static void print_1_to_n(int i, int n) {
        if (i > n) {
            return;
        }

        System.out.println(i);
        print_1_to_n(i + 1, n);
    }
}

File: 02_recursion_basics/02_02_print_1_to_n/print_1_to_n.cpp
#include <bits/stdc++.h>

using namespace std;

void print_1_to_n(int i, int n)
{
    if (i > n)
    {
        return;
    }

    cout << i << endl;
    print_1_to_n(i + 1, n);
}

int main()
{
    int num = 5;
    print_1_to_n(1, num);
}